### ISSUE TITLE
Cannot create provider: incorrect report of conflicting arguments

### ISSUE THREAD

**Issue by sfc-gh-cxu** [pulumi/pulumi-snowflake#248]
### What happened?

This following code should create a snowflake provider without issue. (This code is using [github.com/pulumi/pulumi-snowflake/sdk](http://github.com/pulumi/pulumi-snowflake/sdk) v0.25.0)
```go
	args := &snowflake.ProviderArgs{
		Account:     pulumi.String("snowhouse"),
		Region:      pulumi.String("us-west-2"),
		Role:        pulumi.String("ENGCLOUD_INGEST_RL"),
		Username:    pulumi.String("ENGCLOUD_INGEST_USER"),
		Password:    pulumi.String("<REDACTED>"),
		BrowserAuth: pulumi.Bool(false),
	}

	provider, err := snowflake.NewProvider(p.Ctx, "engcloud-ingest-user-provider", args)
	if err != nil {
		return err
	}
```

but I got 
```
  pulumi:providers:snowflake (engcloud-ingest-user-provider):
    error: rpc error: code = Unknown desc = could not validate provider configuration: 10 errors occurred:
    	* Conflicting configuration arguments: "oauth_access_token": conflicts with browser_auth
    	* Conflicting configuration arguments: "oauth_client_id": conflicts with browser_auth
    	* Conflicting configuration arguments: "oauth_redirect_url": conflicts with browser_auth
    	* Conflicting configuration arguments: "oauth_refresh_token": conflicts with browser_auth
    	* Conflicting configuration arguments: "oauth_client_secret": conflicts with browser_auth
    	* Conflicting configuration arguments: "oauth_endpoint": conflicts with browser_auth
    	* Conflicting configuration arguments: "browser_auth": conflicts with password
    	* Conflicting configuration arguments: "password": conflicts with browser_auth
    	* Conflicting configuration arguments: "private_key_path": conflicts with browser_auth
    	* Conflicting configuration arguments: "private_key_passphrase": conflicts with browser_auth
```

### Expected Behavior

.

### Steps to reproduce

.

### Output of `pulumi about`

```
$ pulumi about                   
CLI
Version      3.67.1
Go Version   go1.20.4
Go Compiler  gc

Plugins
NAME       VERSION
aws        5.28.0
aws        4.38.1
go         unknown
snowflake  0.21.0
vault      4.6.0

Host
OS       darwin
Version  13.3.1
Arch     x86_64

This project is written in go: executable='/usr/local/go/bin/go' version='go version go1.20.2 darwin/amd64
```

### Additional context

I had to workaround this problem by downgrade to v0.21, and to use the default snowflake provider by setting env vars
```
$ SNOWFLAKE_USE_BROWSER_AUTH="false" \
SNOWFLAKE_PASSWORD="<redacted>" \
SNOWFLAKE_USER="ENGCLOUD_INGEST_USER" \
SNOWFLAKE_REGION="us-west-2" \
SNOWFLAKE_ACCOUNT="snowhouse" \
pulumi preview 
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by lukehoban** [comment]
This sounds like it could be caused by the same underlying issue in the terraform bridge tracked in https://github.com/pulumi/pulumi-terraform-bridge/issues/577.

**Comment by t0yv0** [comment]
Thank you for reporting this issue!

The issue you are seeing is activated by the https://github.com/pulumi/pulumi-terraform-bridge/releases/tag/v3.44.0 release that re-enabled validating provider configuration against TF validators in the CheckConfig method. This seems to have uncovered another issue https://github.com/pulumi/pulumi/issues/12971 in the SDK program generator for Go. I'm looking for a path forward to unblock this scenario while keeping the CheckConfig validation enabled as it is beneficial.

So first we will likely need to fix 12971 as I do not see a workaround available for that problem.

After 12971 is fixed the program would still fail with:

```
        * Conflicting configuration arguments: "password": conflicts with browser_auth
        * Conflicting configuration arguments: "browser_auth": conflicts with password
```

The validator is unhappy about `BrowserAuth: false` and would prefer it to be omitted. Commenting out `// BrowserAuth` makes the program work.

I've tried using Transform option on SchemaInfo to coerce false to null in browser_auth at the Pulumi level in the provider. Unfortunately it does not work as of the current version of the bridge - it does transform false to null but that still generates the same conflicting warning from the TF layer.  If we need to make this work we can looking at extending the bridge to support Transform specifying gracefully dropping an input from the surrounding map as opposed to sending a Null map entry.

**Comment by t0yv0** [comment]
Note also that using Terraform provider directly,   browser_auth and password parameters conflict and generate errors: 

```terraform
terraform {
  required_providers {
    snowflake = {
      source  = "Snowflake-Labs/snowflake"
      version = "~> 0.59"
    }
  }
}

provider "snowflake" {
  account = "snowhouse"
  region = "us-west-2"
  role = "ENGCLOUD_INGEST_RL"
  username = "ENGCLOUD_INGEST_USER"
  password = "<REDACTED>"
  browser_auth = false
}

resource "snowflake_user" "u" {
  name = "go-user"
}
```

```
╷
│ Error: Conflicting configuration arguments
│ 
│   with provider["registry.terraform.io/snowflake-labs/snowflake"],
│   on main.tf line 15, in provider "snowflake":
│   15:   password = "<REDACTED>"
│ 
│ "password": conflicts with browser_auth
╵
╷
│ Error: Conflicting configuration arguments
│ 
│   with provider["registry.terraform.io/snowflake-labs/snowflake"],
│   on main.tf line 16, in provider "snowflake":
│   16:   browser_auth = false
│ 
│ "browser_auth": conflicts with password
╵
```

I'm inclined to retain that behavior. Perhaps if we wanted browser_auth=false to not conflict with password, fixing this at upstream Terraform level would be best.

Now for all the other spurious errors this I indeed a bug in Pulumi and a fix is coming shortly in #251  - I will post here once we have a release with that fix.

**Comment by mikhailshilkov** [comment]
@t0yv0 will we run a post-mortem on this regression?

**Comment by t0yv0** [comment]
https://github.com/pulumi/home/issues/2876
